core/controllers/resources_test.py

```diff
@@ -343,6 +344,52 @@ def test_upload_bad_image(self):
 
         self.logout()
 
+    def test_upload_an_invalid_svg_image(self):
+        """Test upload of an invalid SVG image."""
+
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+
+        # Upload an invalid SVG image.
+        response_dict = self.post_json(
+            '%s/exploration/0' % self.IMAGE_UPLOAD_URL_PREFIX,
+            {'filename': 'test.svg'},
+            csrf_token=csrf_token,
+            expected_status_int=400,
+            upload_files=(('image', 'unused_filename', '<badsvg></badsvg>'),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'],
+            'Unsupported tags/attributes found in the SVG:\ntags: '
+            '[u\'badsvg\']\n')
+
+        self.logout()
+
+    def test_upload_a_large_svg(self):
+        """Test upload of an invalid SVG image."""
```

Comment:
You probably want a more specific docstring here; this is identical to the previous one.